storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -526,6 +526,10 @@ public boolean isJustifiedCheckpointFullyValidated() {
         && store.getForkChoiceStrategy().isFullyValidated(store.getJustifiedCheckpoint().getRoot());
   }
 
+  public Optional<UInt64> getLatestValidFinalizedSlot() {
+    return store == null ? Optional.empty() : Optional.of(store.getLatestValidFinalizedSlot());
+  }
```

Comment:
I don't think we need this to return `Optional` since we know that the genesis block is always fully valid, if we don't have a `store` yet we can just return `ZERO` and keep things simple.